akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/BasicDirectivesSpec.scala

```diff
@@ -107,4 +108,35 @@ class BasicDirectivesSpec extends RoutingSpec {
       } ~> check { responseAs[String] shouldEqual "" }
     }
   }
-}
\ No newline at end of file
+
+  "The `extractStrictEntity` directive" should {
+    "change request to contain strict entity for inner routes" in {
+      val chunks = () ⇒ List("Akka", "HTTP").map(HttpEntity.Chunk(_)).iterator
+      val entity = HttpEntity.Chunked(ContentTypes.`text/plain(UTF-8)`, Source.fromIterator(chunks))
+
+      Post("/abc", entity) ~> {
+        extractRequestEntity { before ⇒
+          extractStrictEntity(200.millis) { _ ⇒
+            extractRequestEntity { after ⇒
+              complete(Seq(before, after).map(_.isInstanceOf[HttpEntity.Strict]).mkString(" => "))
+            }
+          }
+        }
+      } ~> check { responseAs[String] shouldEqual "false => true" }
+    }
+
+    "only consume data once when nested" in {
+      val random = java.util.concurrent.ThreadLocalRandom.current
+      val chunks = Source.fromIterator(() ⇒ Iterator.continually(random.nextInt(0, 2).toString).take(100).map(HttpEntity.Chunk(_)))
+      val entity = HttpEntity.Chunked(MediaTypes.`application/x-www-form-urlencoded`.toContentType(HttpCharsets.`UTF-8`), chunks)
```

Comment:
I can simplify this to use `text/plain`. The form-urlencoded is a left-over from trying to come up with a reproducer. Finally, using random number was the only way I found to check if the request data was read twice.